packages/expo-linking/src/createURL.ts

```diff
@@ -136,20 +140,27 @@ export function createURL(
 export function parse(url: string): ParsedURL {
   validateURL(url);
 
-  const parsed = URL(url, /* parseQueryString */ true);
-
-  for (const param in parsed.query) {
-    parsed.query[param] = decodeURIComponent(parsed.query[param]!);
+  const queryParams: Record<string, string> = {};
+  let path: string | null = null;
+  let hostname: string | null = null;
+  let scheme: string | null = null;
+
+  try {
+    const parsed = new URL(url);
+
+    parsed.searchParams.forEach((value, key) => {
+      queryParams[key] = decodeURIComponent(value);
```

Comment:
I opened ENG-10487 since this looks like a bug that was present in the old version of this library too.